### DISCUSSION TITLE
Query / KQL Question - How to hide preservation libraries and sitepages/templates/ from appearing in results

### DISCUSSION THREAD

**Discussion by bkvwilson** [microsoft-search/pnp-modern-search#1708] [General]
Hi, 

We are seeing items in the **"PreservationHoldLibrary"** (we have retention policies on all sites in SharePoint Online) appearing in the search result. We also have **site templates** that are appearing incorrectly in our result set. Any suggestions on good practice to prevent these appearing in search results list?

Lets say the query template to return all pages in a set of sites is as follows:

_{searchTerms} 
Path:"https://tenantname.sharepoint.com/sites/siteprefixone*"
Path:"https://tenantname.sharepoint.com/sites/siteprefixtwo*"
FileType:aspx_

Should we use -Path:*PreservationHoldLibrary* and -Path:*/SitePages/Templates/* to exclude these libraries or is there a better way to achieve this? 

Thanks
Brian

**Comment by wobba** [comment]
Use `-ParentLink:PreservationHoldLibrary`. And yes, you need to find a filter which exclude what you don't want. `Path` does not support partial matching. So for template libraries it's a bit harder as ParentLink might not work 100%.

**Reply by mikezimm** [reply]
@wobba , 
You mentioned Path does not support partial matching.

I'm not sure if this is what they were referring to but when I was testing, I found it did do what I would call partial matching of the path.

In my case, I was testing this query:
{SearchBoxQuery}* path:https://tenant.sharepoint.com/sites/WebPartDev

Note my Site url was:  https://tenant.sharepoint.com/sites/WebPartDev

It  returned all items in the site including list items in the site (well I didn't count them all but looked pretty complete)

It also seemed to work even if I used something like this (partial path):
{SearchBoxQuery}* path:https://tenant.sharepoint.com/sites/WebPar

It would still find all the items (including list items) in the site.

**Reply by wobba** [reply]
Path support pre-fix matching, mathing everything from https:// up to what you type, but not substring partial matching. That is what I meant. So `path:WebPartDev` won't work, while `ParentLink:WebPartDev` can.
